operators/tokenizer/bpe_tokenizer.hpp

```diff
@@ -80,8 +86,62 @@ class BpeModel {
 
     id2token_map_.resize(vocab_map_.size());
     for (const auto& [t, i] : vocab_map_) {
+      if (i > static_cast<uint32_t>(std::numeric_limits<int32_t>::max())) {
+        continue;  // safe purpose.
+      }
+      if (i > id2token_map_.size()) {
+        id2token_map_.resize(i + 1);
+      }
       id2token_map_[i] = t;
     }
+
+    return nullptr;
+  }
+
+  OrtStatusPtr LoadAddedTokens(const char* added_tokens) {
```

Comment:
is there some HF reference for this?

Reply:
yes, check here, https://github.com/huggingface/transformers/blob/cc3e4781854a52cf090ffde28d884a527dab6708/src/transformers/tokenization_utils.py#L52,